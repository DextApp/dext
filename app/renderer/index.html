<!doctype html>
<html>
  <head>
    <title>Dext</title>
    <style type="text/css">
      body,
      html {
        margin: 0;
        padding: 0;
        background-color: transparent;
      }
      #app {
        background-color: transparent;
      }
    </style>
  </head>

  <body>

    <div id="app"></div>

    <script type="text/javascript" src="lib/bundle.js"></script>

  </body>
</html>
