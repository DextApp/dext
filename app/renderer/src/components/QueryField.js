import React, { Component, PropTypes } from 'react';
import { compose, style } from 'glamor';
import { ThemeSchema } from '../schema';

const base = style({
  padding: 15,
});

const search = style({
  backgroundColor: 'transparent',
  border: 0,
  boxSizing: 'border-box',
  fontSize: 50,
  marginBottom: 0,
  paddingBottom: 15,
  outline: 0,
  overflow: 'visible',
  width: '100%',
  height: 50,
});

const QueryField = class extends Component {
  render() {
    const { theme, onChange, value } = this.props;
    const styles = theme && theme.search
      ? compose(search, theme.search)
      : search;

    return (
      <div {...base}>
        <input
          ref={c => { this.input = c && c; }}
          onChange={onChange}
          value={value}
          {...styles}
        />
      </div>
    );
  }
};

QueryField.propTypes = {
  onChange: PropTypes.func,
  theme: ThemeSchema,
  value: PropTypes.string,
};

export default QueryField;
