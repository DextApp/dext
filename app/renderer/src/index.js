/* global document */
import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import DevTools from './components/DevTools';
import AppContainer from './containers/AppContainer';
import configureStore from './store';

const store = configureStore();

// load the dev tools if necessary
let devTools = '';
if (process.env.NODE_ENV === 'development') {
  devTools = <DevTools />;
}

const DextApp = () => (
  <Provider store={store}>
    <div>
      <AppContainer />
      {devTools}
    </div>
  </Provider>
);

const rootNode = document.getElementById('app');

ReactDOM.render(<DextApp />, rootNode);
