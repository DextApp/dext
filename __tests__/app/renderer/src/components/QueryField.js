import React from 'react';
import renderer from 'react-test-renderer';
import QueryField from '../../../../../app/renderer/src/components/QueryField';

describe('<QueryField /> component', () => {
  let theme = {};

  beforeEach(() => {
    theme = {};
  });

  it('should render a query field component with no value', () => {
    const component = renderer.create(
      <QueryField theme={theme} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a query field component with some value', () => {
    const component = renderer.create(
      <QueryField theme={theme} value={'TEST'} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
