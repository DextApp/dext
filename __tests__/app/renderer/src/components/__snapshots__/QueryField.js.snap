exports[`<QueryField /> component should render a query field component with no value 1`] = `
<div
  data-css-1ykjhyy="">
  <input
    data-css-bj9j4d=""
    onChange={[Function]}
    value="" />
</div>
`;

exports[`<QueryField /> component should render a query field component with some value 1`] = `
<div
  data-css-1ykjhyy="">
  <input
    data-css-bj9j4d=""
    onChange={[Function]}
    value="TEST" />
</div>
`;
